osu.Game/Screens/Ranking/Expanded/StarRatingDisplay.cs

```diff
@@ -47,9 +61,20 @@ private void load(OsuColour colours, BeatmapDifficultyCache difficultyCache)
                 ? ColourInfo.GradientVertical(Color4Extensions.FromHex("#C1C1C1"), Color4Extensions.FromHex("#595959"))
                 : (ColourInfo)colours.ForDifficultyRating(difficulty.DifficultyRating);
 
+            colorContainer.Colour = backgroundColour;
+
+            wholePartText.Text = $"{wholePart}";
+            fractionPartText.Text = $"{separator}{fractionPart}";
```

Comment:
These changes have removed some fade transitions that were naturally there due to how the component was originally structured (everything in a buffered container), and therefore this looks worse now. To understand what I mean try toggling mods on `master` and this branch (so that the colour of the star rating changes).

Not sure whether it's worth holding this pull up for that as I understand a redesign of the song select screen is in the works, but yeah.

Reply:
Hmm, that's true. Things that come to my mind to restore the previous behavior on rating change:
- Fade out sprite texts and set alpha back to 100% (don't know if it's a thing to do smth like that)
- Fade out whole text container and set alpha back
    - Both would have the problem that the fading is set inside `StarRatingDisplay` and therefore not possible to set it from outside
-  Fade out the top right metadata container (Containing `StarRatingDisplay` and `StatusPill`) in `BeatmapWedgeInfo` and recreate it
- Fade out and recreate the whole Info container

or your last point `ignore it for now`. I don't know what the best way is to handle this case

Reply:
Would probably be best to incorporate the transition into the component itself:

* Background would simply use `.FadeColour()`
* As for the foreground, you'd probably need to `.FadeOut().Expire()` the old sprite text and create a new one with `.FadeIn()` to preserve the visual effect.

Might be a bit finicky in places because transforms cannot be added in `load()`, so you'd need some `IsLoaded` checks. It's a pretty rudimentary pattern though, should find many usages around.

Reply:
`.FadeOut(250).Expire()` results in a weird transition where the size of the container gets larger and smaller:
https://streamable.com/1cgb0z
I implemented it now w/o the expire and it feels somewhat good. Here is a comparison with the transition on master:
https://streamable.com/3o1nv1

Reply:
The expire was supposed to be an integral part of this. It was to make sure the old texts are cleaned up after the fades have completed. Have you checked whether they really are, using the draw visualiser?

Reply:
ohh, i misswrote i meant i did it now without the fadeout the expire is still there and used